src/Features/ShippingLabelBanner.php

```diff
@@ -36,6 +36,8 @@ public function __construct() {
 			return;
 		}
 		add_action( 'add_meta_boxes', array( $this, 'add_meta_boxes' ), 6, 2 );
+		add_filter( 'woocommerce_components_settings', array( $this, 'component_settings' ), 20 );
```

Comment:
`woocommerce_components_settings` Is actually deprecated in WC 2.5


There's also a note on `woocommerce_shared_settings`
> Developers, do not use this hook as it is likely to be removed.
Instead, use the data api:
Automattic\WooCommerce\Blocks\Package::container()
    ->get( Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry::class )
    ->add( $key, $value )
So maybe we should do that?

Also probably if we could do it on `meta_box()` or sometime after `should_show_meta_box()` check so we only included it on the pages it is needed.

Reply:
I can replace this to something like
```
$data_registry = Package::container()->get(AssetDataRegistry::class);
if (onboarding not loaded) {
    $data_registry->add( 'onboarding', $settings['onboarding'] );
} else {
    // use onboarding's setting
    $data_registry-><some function to retreieve onboarding data>();
}
```
I couldn't find a way to get the existing settings. Still poking through the `AssetDataRegistry` class, do you know a way to get that please?

Reply:
Added follow up here: https://github.com/Automattic/woocommerce-shipping-issues/issues/28